### ISSUE TITLE
linter: `no-noninteractive-tabindex` does not error when the tabIndex value is an expression

### ISSUE THREAD

**Issue by JasonKaz** [oxc-project/oxc#22431]
### What version of Oxlint are you using?

1.64.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["jsx-a11y"],
  "rules": {
    "jsx-a11y/no-noninteractive-tabindex": "error"
  }
}
```

### What happened?

If the `tabIndex` value is an expression it is not evaluated, so the rule does not trigger. This differs from the eslint implemenation.

Repro can be found [here](https://playground.oxc.rs/?lintRules=jsx-a11y%2Fno-noninteractive-tabindex&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=import+React%2C+%7B+useEffect%2C+useRef+%7D+from+%27react%27%0A%0Ainterface+Props+%7B%0A++isTabbable%3A+boolean%3B%0A%7D%0A%0Aconst+DummyComponent%3A+React.FC+%3D+%28props%3A+Props%29+%3D%3E+%7B%0A++%2F%2F+This+should+violate+no-noninteractive-tabindex%0A++%2F%2F+because+the+value+will+result+in+an+integer%0A++return+%28%0A++++%3Cdiv+tabIndex%3D%7Bprops.isTabbable+%3F+1+%3A+0%7D%3E%3C%2Fdiv%3E%0A++%29%0A%7D%0A%0Aconst+DummyComponen2%3A+React.FC+%3D+%28%29+%3D%3E+%7B%0A++return+%28%0A++++%3Cdiv+tabIndex%3D%7B0%7D%3E%3C%2Fdiv%3E%0A++%29%0A%7D).
